SetupDataPkg/Tools/CommonUtility.py

```diff
@@ -97,3 +99,32 @@ def array_str_to_value(val_str):
         each = each.strip()
         value = (value << 8) | int(each, 0)
     return value
+
+
+def get_xml_full_hash(xml_path):
+    # Parse the XML file
+    tree = ET.parse(xml_path)
+    root = tree.getroot()
+
+    # List to store all tags, attributes, and values
+    hash_content = []
+
+    def traverse(element):

```

Comment:
Why do we traverse the node instead of hashing the entire xml file? Wouldn't that be simpler?

Reply:
this is to prevent any one accidently adding space in the xml files or using linux to edit the xml file (line break btwn windows and linux are different)